app/main.py

```diff
@@ -1,25 +1,77 @@
+from __future__ import annotations
+
 from abc import ABC
+from typing import Any
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: Any, name: str) -> None:
+        self.public_name = name
+        self.protected_name = "_" + name
+
+    def __get__(self, obj: Any, objtype: Any = None) -> None:
+        return getattr(obj, self.protected_name)
+
+    def __set__(self, obj: Any, value: Any) -> None:
+        if not isinstance(value, int):
+            raise TypeError("Value should be integer.")
+
+        if not self.min_amount <= value <= self.max_amount:
+            raise ValueError(
+                f"Value should more than {self.min_amount}"
+                f" and less than {self.max_amount}."
+            )
+
+        setattr(obj, self.protected_name, value)
 
 
 class Visitor:
-   pass
+    def __init__(self, name: str, age: int, weight: int, height: int) -> None:
+        self.name = name
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
 class SlideLimitationValidator(ABC):
-    pass
+
+    def __init__(self, age: int, weight: int, height: int) -> None:
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
 class ChildrenSlideLimitationValidator(SlideLimitationValidator):
-    pass
+
+    age = IntegerRange(4, 14)
+    weight = IntegerRange(20, 50)
+    height = IntegerRange(80, 120)
 
 
 class AdultSlideLimitationValidator(SlideLimitationValidator):
-    pass
+
+    age = IntegerRange(14, 60)
+    weight = IntegerRange(50, 120)
+    height = IntegerRange(120, 220)
 
 
 class Slide:
-    pass
+    def __init__(
+            self,
+            name: str,
+            limitation_class:
+            type[SlideLimitationValidator]
+    ) -> None:
+        self.name = name
+        self.limitation_class = limitation_class
+
+    def can_access(self, visitor: Visitor) -> bool:
+        try:
+            self.limitation_class(visitor.age, visitor.weight, visitor.height)
+            return True
+        except ValueError or TypeError:
```

Comment:
It is preferable to catch multiple errors in this way
```suggestion
        except (ValueError, TypeError):
```